mqtt-client-Node.js/index.js

```diff
@@ -1,71 +1,125 @@
+// const { connectOptions } = require('./use_ws.js')
+const { connectOptions } = require('./use_wss.js')
 
-const program = new Command()
-program
-  .option('-p, --protocol <type>', 'connect protocol: mqtt, mqtts, ws, wss. default is mqtt', 'mqtt')
-  .parse(process.argv)
+/**
+ * this demo uses EMQX Public MQTT Broker (https://www.emqx.com/en/mqtt/public-mqtt5-broker), here are the details:
+ *
+ * Broker host: broker.emqx.io
+ * TCP Port: 1883
+ * SSL/TLS Port: 8883
+ * WebSocket port: 8083
+ * WebSocket over TLS/SSL port: 8084
+ */
 
-const host = 'broker.emqx.io'
-const port = '1883'
-const clientId = `mqtt_${Math.random().toString(16).slice(3)}`
-
-// connect options
-const OPTIONS = {
+const clientId = 'emqx_nodejs_' + Math.random().toString(16).substring(2, 8)
+const options = {
   clientId,
   clean: true,
   connectTimeout: 4000,
-  username: 'emqx',
-  password: 'public',
+  /**
+   * By default, EMQX allows clients to connect without authentication.
+   * https://docs.emqx.com/en/enterprise/v4.4/advanced/auth.html#anonymous-login
+   */
+  username: 'emqx_test',
+  password: 'emqx_test',
   reconnectPeriod: 1000,
+  // for more options and details, please refer to https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options
 }
-// protocol list
-const PROTOCOLS = ['mqtt', 'mqtts', 'ws', 'wss']
 
-// default is mqtt, unencrypted tcp connection
-let connectUrl = `mqtt://${host}:${port}`
-if (program.protocol && PROTOCOLS.indexOf(program.protocol) === -1) {
-  console.log('protocol must one of mqtt, mqtts, ws, wss.')
-} else if (program.protocol === 'mqtts') {
-  // mqtts， encrypted tcp connection
-  connectUrl = `mqtts://${host}:8883`
-  OPTIONS['ca'] = fs.readFileSync('./broker.emqx.io-ca.crt')
-} else if (program.protocol === 'ws') {
-  // ws, unencrypted WebSocket connection
-  const mountPath = '/mqtt' // mount path, connect emqx via WebSocket
-  connectUrl = `ws://${host}:8083${mountPath}`
-} else if (program.protocol === 'wss') {
-  // wss, encrypted WebSocket connection
-  const mountPath = '/mqtt' // mount path, connect emqx via WebSocket
-  connectUrl = `wss://${host}:8084${mountPath}`
-  OPTIONS['ca'] = fs.readFileSync('./broker.emqx.io-ca.crt')
-} else {}
+const { protocol, host, port } = connectOptions
+/**
+ * if protocol is "ws", connectUrl = "ws://broker.emqx.io:8083/mqtt"
+ * if protocol is "wss", connectUrl = "wss://broker.emqx.io:8084/mqtt"
+ *
+ * for more details about "mqtt.connect" method & options,
+ * please refer to https://github.com/mqttjs/MQTT.js#mqttconnecturl-options
+ */
+let connectUrl = `${protocol}://${host}:${port}`
+if (['ws', 'wss'].includes(protocol)) {
+  // mqtt: MQTT-WebSocket uniformly uses /path as the connection path,
+  // which should be specified when connecting, and the path used on EMQX is /mqtt.
+  connectUrl += '/mqtt'
+}
+if (['mqtts', 'wss'].includes(protocol)) {
+  options['ca'] = fs.readFileSync('./broker.emqx.io-ca.crt')
```

Comment:
does it work without CA?

Reply:
Yes. I noticed that the original demo and other backend code had CA, so I kept it.

Reply:
you need test (use EMQX Cloud Deployment), and add some notes there.